src/ui/bitmapFont/BitmapFontTextViewBacking.js

```diff
@@ -838,15 +846,42 @@ export default class BitmapFontTextViewBacking {
     return baseline * fontSizeScale;
   }
 
-  get text() {
-    return this._text;
+  set autoSize(value) {
+    if (this._autoSize !== value) {
+      this._autoSize = value;
+
+      this.invalidate();
+    }
+  }
+
+  get autoSize() {
+    return this._autoSize;
   }
 
   set text(value) {
     value = '' + value;
-    if (this._text === value) { return; }
-    this._text = value;
-    this.invalidate();
+
+    if (this._text !== value) {
+      this._text = value;
+      this._listener._opts.size = this._baseSize;
+
+      this.invalidate();
+
+      if (this._autoSize && this._baseSize && this._listener._opts.font) {
+        let viewWidth = this._listener._opts.width;
```

Comment:
Read view style width for views with dynamic size?  Makes me think actually, what is the desired behavior with a dynamically sized view? Maybe the font scaling logic should live at a view level? Or at least make a public autoSize method